LogMonitor/src/LogMonitor/Main.cpp

```diff
@@ -25,6 +25,27 @@ std::unique_ptr<EventMonitor> g_eventMon(nullptr);
 std::vector<std::shared_ptr<LogFileMonitor>> g_logfileMonitors;
 std::unique_ptr<EtwMonitor> g_etwMon(nullptr);
 
+/// Handle signals.
+///
+/// \return None
+///
+void signalHandler(int signum) {
+    logWriter.TraceError(
+        Utility::FormatString(L"Catastrophic failure! Signal received: %d", signum).c_str()
+    );
+
+    // Terminate the program
+    std::exit(signum);
+}
+
+/// Set up a signal handler
+///
+/// \return None
+///
+void setSignalHandler(int signum, void (*handler)(int)) {

```

Comment:
Nit: Why not `std::signal(signum, signalHandler);` to avoid passing the handler every time we call `setSignalHandler()`

Reply:
Using `std::signal(signum, signalHandler)` directly in the `setSignalHandler()` function would indeed avoid passing the handler every time we call `setSignalHandler()`. However, it is generally considered a good practice to encapsulate the signal handling logic within a function, like `setSignalHandler()`, to provide a clear and modular interface for setting up signal handlers. This allows for better code organization and separation of concerns. Additionally, encapsulating the signal handling logic in a separate function allows for easier modification or extension of the signal handling behavior in the future.